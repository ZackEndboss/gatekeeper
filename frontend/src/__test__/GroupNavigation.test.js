import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import {MockStore} from './mockStore';
import {Provider} from 'react-redux';
import ConnectedGroupNavigation, {GroupNavigation} from '../GroupNavigation';

jest.mock('../Group', () => 'Group');

describe('GroupNavigation component', () => {
    it('render', () => {
        const component = renderer.create(<GroupNavigation groups={['group1', 'group2']} active_group={0}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should contain a tab for every group', () => {
        const component = shallow(<GroupNavigation groups={['group1', 'group2']} active_group={0}/>);
        expect(component.find('Tab').length).toEqual(2);
    });
});

describe('load from state', () => {
    it('should load the state correctly', () => {
        const storeMock = MockStore({
            'tabs': {
                active_tab: 0
            },
            'gates': {
                'ftx': {},
                'fty': {}
            }
        });

        const subject = (
            <Provider store={storeMock}>
                <ConnectedGroupNavigation/>
            </Provider>);
        const component = renderer.create(subject);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
