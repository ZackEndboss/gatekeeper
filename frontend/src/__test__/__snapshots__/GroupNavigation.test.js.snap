// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupNavigation component render 1`] = `
<div
  className={undefined}
  id="controlled"
  style={undefined}
>
  <ul
    className="nav nav-tabs"
    role="tablist"
  >
    <li
      className="active"
      role="presentation"
      style={undefined}
    >
      <a
        aria-controls="controlled-pane-0"
        aria-selected={true}
        href="#"
        id="controlled-tab-0"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex={undefined}
      >
        group1
      </a>
    </li>
    <li
      className=""
      role="presentation"
      style={undefined}
    >
      <a
        aria-controls="controlled-pane-1"
        aria-selected={false}
        href="#"
        id="controlled-tab-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex={-1}
      >
        group2
      </a>
    </li>
  </ul>
  <div
    className="tab-content"
  >
    <div
      aria-hidden={false}
      aria-labelledby="controlled-tab-0"
      className="tab-pane active fade in"
      id="controlled-pane-0"
      role="tabpanel"
    >
      <Group
        group="group1"
      />
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="controlled-tab-1"
      className="tab-pane fade"
      id="controlled-pane-1"
      role="tabpanel"
    >
      <Group
        group="group2"
      />
    </div>
  </div>
</div>
`;

exports[`load from state should load the state correctly 1`] = `
<div
  className={undefined}
  id="controlled"
  style={undefined}
>
  <ul
    className="nav nav-tabs"
    role="tablist"
  >
    <li
      className="active"
      role="presentation"
      style={undefined}
    >
      <a
        aria-controls="controlled-pane-0"
        aria-selected={true}
        href="#"
        id="controlled-tab-0"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex={undefined}
      >
        ftx
      </a>
    </li>
    <li
      className=""
      role="presentation"
      style={undefined}
    >
      <a
        aria-controls="controlled-pane-1"
        aria-selected={false}
        href="#"
        id="controlled-tab-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex={-1}
      >
        fty
      </a>
    </li>
  </ul>
  <div
    className="tab-content"
  >
    <div
      aria-hidden={false}
      aria-labelledby="controlled-tab-0"
      className="tab-pane active fade in"
      id="controlled-pane-0"
      role="tabpanel"
    >
      <Group
        group="ftx"
      />
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="controlled-tab-1"
      className="tab-pane fade"
      id="controlled-pane-1"
      role="tabpanel"
    >
      <Group
        group="fty"
      />
    </div>
  </div>
</div>
`;
